### PR TITLE
Rails 7 Upgrade Main Branch

### PR DESCRIPTION
**Pull request by kaylareopelle**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Main branch for developing features to provide Rails 7.0 support. Please create new branches for the milestone off of `test/rails-7-upgrade` and merge them into this branch until all test suites pass.

This branch currently runs tests against [Rails 7.0.0.rc1](https://rubygems.org/gems/rails/versions/7.0.0.rc1)

# Related Github Issue
Related Milestone: https://github.com/newrelic/newrelic-ruby-agent/milestone/21 

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
Github Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,10 +61,10 @@ jobs:
...
               "2.7.5": {
-                "rails": "norails,rails61,rails60"
+                "rails": "norails,rails61,rails60,rails70"
               },
               "3.0.3": {
-                "rails": "norails,rails61,rails60"
+                "rails": "norails,rails61,rails60,rails70"
               },
               "3.1.0": {
                 "rails": "norails,rails61"
```
Comment (tannalynn) [inline .github/workflows/ci.yml:70]: We should add rails70 to 3.1.0 also.

**Comment by kaylareopelle** [comment]
Now that all the tests are passing, would folks be open to merging this into dev and fixing the last issue off of that branch?

**Review by tannalynn — APPROVED** [review]
LGTM! 👍

**Comment by codeodor** [comment]
Is it safe to pin our gemfile to the `dev` branch so we can upgrade our app to Rails 7 now, or do we need to wait for the release of a new gem version?

**Comment by kaylareopelle** [comment]
Hi @codeodor! 

It should be safe to pin your Gemfile to the `dev` branch to upgrade to Rails 7 now. We'd appreciate some customers testing things before the new release so that we can find out if there are any areas we missed. As a head's up, there is one warning our team is still addressing related to ActiveStorage that may cause the timing for some `Nested/Controller/active_support/*` segments to be inaccurate (#897). 

If you choose to upgrade, we'd love to hear how it goes!

Thanks for reaching out! 🚀

**Comment by fallwith** [comment]
Hello @codeodor, the changes from this PR have now gone out to RubyGems in the new v8.4.0 release. Please point now to the v8.4.0 gem and not to dev anymore, as dev may soon receive other changes that you might not want / might be unstable.

**Comment by codeodor** [comment]
Sorry, I had to drop our upgrade for a while, so I didn't get a chance to test out the new gem before you released, but I'm back on it now and the app boots with the new one, so thank you!!! I appreciate y'alls work on it.

**Pull request merged**